### PR TITLE
Cilium Custom Go Vet Tool

### PR DESCRIPTION
**Pull request by nathanjsweet**
The first use of this tool will be to ensure that
the `time.After` method is not used in a for loop

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/cilium/customvet
+
+go 1.15
+
+require (
+	github.com/aws/aws-sdk-go v1.36.3
```
Comment (christarazi) [inline go.mod:6]: Where did this come from?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	golang.org/x/tools v0.0.0-20201207204333-a835c872fcea
```
Comment (christarazi) [inline go.mod:7]: Should we be vendoring in our dependencies?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package main
+
+import (
+	"github.com/cilium/customvet/analysis"
+	"golang.org/x/tools/go/analysis/singlechecker"
```
Comment (christarazi) [inline main.go:5]: I imagine we want to adhere to the same code style guidelines as the main Cilium codebase so import ordering here should be:

```
import (
  std

  cilium/customvet

  3rd party
)
```

File: analysis/analysis.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		}
+	})
+	return nil, nil
+}
+
+type timeAfterCheck struct {
+	aliases []string
+}
+
+func (t *timeAfterCheck) check(n ast.Node) (ok bool, report string, pos token.Pos) {
```
Comment (christarazi) [inline analysis/analysis.go:64]: Maybe we can start to move these individual checks into their own files. Would be good to have structure as we add more checks.

**Review by christarazi — COMMENTED** [review]
This is pretty awesome! Didn't know it was possible to do this without upstreaming the check into `govet`. 

Also, I imagine we need to license this code as well.

A few questions below and inlined:
Is this intended to be integrated into our GH Actions job? How do we use it?

**Comment by nathanjsweet** [comment]
> Is this intended to be integrated into our GH Actions job? How do we use it?

Yes it is as simple as running `go vet --tool=$(which customvet) ./...`.
I was thinking it could be [with the other prechecks](https://github.com/cilium/cilium/blob/8030a81dbbcc189814163742fdb32af740c2cb80/Makefile#L501).

**Review by christarazi — APPROVED** [review]
:rocket:

**Pull request merged**